### PR TITLE
Implement build configuration header for SRT_ENABLE_AEAD_API_PREVIEW.

### PR DESCRIPTION
**Pull request by jlsantiago0**
Implement solution for https://github.com/Haivision/srt/issues/2610 FR request.

### CODE REVIEW

**Comment by jlsantiago0** [comment]
This is a pattern that can be used to add build configuration options that effect the library ABI. Currently some of the them like ENABLE_AEAD_API_PREVIEW have to be set by the library  user in order to get the right ABI/API for the library.

As indicated by #2610 this is error prone and dangerous. 

If this PR is accepted, then going forward any ABI affecting configuration options used to build the SRT library can be specified here so that they are set correctly for any subsequent user of the library.

**Comment by jlsantiago0** [comment]
I dont know what other configuration options are needed at this point. ENABLE_BONDING looks suspicous in srt.h.

**Comment by maxsharabayko** [comment]
Looks like this would require raising the minimum CMake version to 3.0.

**Comment by jlsantiago0** [comment]
> Looks like this would require raising the minimum CMake version to 3.0.

Oh. Did I use something that is from CMake 3.0 or later?

**Comment by maxsharabayko** [comment]
> > Looks like this would require raising the minimum CMake version to 3.0.
> 
> Oh. Did I use something that is from CMake 3.0 or later? I thought it was only features that were already being used. For instance to generate the version.h file.

Nope, my bad. I thought `configure_file` is only available starting from CMake 3.0, but it is also available in 2.8.12.
https://cmake.org/cmake/help/v2.8.12/cmake.html#command:configure_file

**Comment by ethouris** [comment]
Why can't this be done the same way as for all other such options - by adding -D to the "definitions" in SRT in cmakefile?

File: CMakeLists.txt [hunk 1]
```diff
@@ -949,10 +949,13 @@ if(DEFINED ENV{TEAMCITY_VERSION})
...
 configure_file("srtcore/version.h.in" "version.h" @ONLY)
-
 list(INSERT HEADERS_srt 0 "${CMAKE_CURRENT_BINARY_DIR}/version.h")
 include_directories("${CMAKE_CURRENT_BINARY_DIR}")
 
+set(SRT_ENABLE_AEAD_API_PREVIEW ${ENABLE_AEAD_API_PREVIEW})
+configure_file("srtcore/srt_config.h.in" "srt_config.h" @ONLY)
+list(APPEND HEADERS_srt "${CMAKE_CURRENT_BINARY_DIR}/srt_config.h")
+
 add_library(srt_virtual OBJECT ${SOURCES_srt} ${SOURCES_srt_extra} ${HEADERS_srt} ${SOURCES_haicrypt} ${SOURCES_common})
```
Comment (ethouris) [inline CMakeLists.txt:959]:
```suggestion
add_library(srt_virtual OBJECT ${SOURCES_srt} ${SOURCES_srt_extra} ${HEADERS_srt} ${SOURCES_haicrypt} ${SOURCES_common})

set_if(SRT_ENABLE_AEAD_API_PREVIEW ENABLE_AEAD_API_PREVIEW)
target_compile_definitions(srt_virtual PRIVATE SRT_ENABLE_AEAD_API_PREVIEW=${SRT_ENABLE_AEAD_API_PREVIEW})
```

File: apps/socketoptions.hpp [hunk 2]
```diff
@@ -255,7 +255,7 @@ const SocketOption srt_options [] {
     { "bindtodevice", 0, SRTO_BINDTODEVICE, SocketOption::PRE, SocketOption::STRING, nullptr},
 #endif
     { "retransmitalgo", 0, SRTO_RETRANSMITALGO, SocketOption::PRE, SocketOption::INT, nullptr }
-#ifdef ENABLE_AEAD_API_PREVIEW
+#if defined(SRT_ENABLE_AEAD_API_PREVIEW) && (SRT_ENABLE_AEAD_API_PREVIEW == 1)
```
Comment (ethouris) [inline apps/socketoptions.hpp:258]:
```suggestion
#if SRT_ENABLE_AEAD_API_PREVIEW
```

**Comment by jlsantiago0** [comment]
> Why can't this be done the same way as for all other such options - by adding -D to the "definitions" in SRT in cmakefile?

Because you need to know how the library was originally compiled. so this information should be stored in generated header files that reflect how the library was built. Otherwise you have no way of knowing if SRT_ENABLE_API_PREVIEW and other configurations that are used to effect the how the library was built and can in fact effect the ABI of the library such that you have to set them correctly when using the library or risk the code using it break. This information needs to be in generated header files. and nowhere else.

**Pull request closed without merging**